dhcpv4/dhcpv4.go

```diff
@@ -493,12 +491,20 @@ func (d *DHCPv4) ToBytes() []byte {
 
 	var sname [64]byte
 	copy(sname[:], []byte(d.ServerHostName))
-	sname[len(d.ServerHostName)] = 0
+	if len(d.ServerHostName) < len(sname) {
+		sname[len(d.ServerHostName)] = 0
+	} else {
+		sname[len(sname)-1] = 0
```

Comment:
actually, you could simplify this whole thing to

sname[len(sname)-1] = 0

sname is pre-initialized to 0 values. the only case where you have to append a 0 to null-terminate a string is if d.ServerHostName > 64.

Reply:
I think I got a slightly simpler idea, relying on the fact that the buffer is 0-initialized. Thanks for reminding me :)